apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsSettings/index.tsx

```diff
@@ -3,46 +3,36 @@ import ButtonV2 from "~/renderer/components/Button";
 import { useTranslation } from "react-i18next";
 import {
   defaultFeatures,
+  groupedFeatures,
   useFeature,
   useFeatureFlags,
 } from "@ledgerhq/live-common/featureFlags/index";
-import { Input, Icons, Flex, SearchInput, Alert, Tag } from "@ledgerhq/react-ui";
+import { Flex, SearchInput, Alert, Tag } from "@ledgerhq/react-ui";
 import { SettingsSectionRow as Row } from "../../../SettingsSection";
 import { FeatureId } from "@ledgerhq/types-live";
-import { InputRenderLeftContainer } from "@ledgerhq/react-ui/components/form/BaseInput/index";
 import { includes, lowerCase, trim } from "lodash";
 import { withV3StyleProvider } from "~/renderer/styles/StyleProviderV3";
 import FeatureFlagDetails from "./FeatureFlagDetails";
 
 const addFlagHint = `\
 If a feature flag is defined in the targeted Firebase environment \
-but it is missing from the following list, you can type its name in \
-the input field below and it will appear in the list. Type the \
+but it is missing from the following list, you can type its **exact** name in \
+the search input and it will appear in the list. Type the \
 flag name in camelCase without the "feature" prefix.\
 `;
 
 export const FeatureFlagContent = withV3StyleProvider((props: { visible?: boolean }) => {
   const { t } = useTranslation();
   const [focusedName, setFocusedName] = useState<string | undefined>();
-  const [hiddenFlagName, setHiddenFlagName] = useState("");
   const [searchInput, setSearchInput] = useState("");
-
-  const trimmedHiddenFlagName = trim(hiddenFlagName);
+  const searchInputTrimmed = trim(searchInput);
 
   const featureFlags = useMemo(() => {
     const featureKeys = Object.keys(defaultFeatures);
-    if (trimmedHiddenFlagName && !featureKeys.includes(trimmedHiddenFlagName))
-      featureKeys.push(trimmedHiddenFlagName);
+    if (searchInputTrimmed && !featureKeys.includes(searchInputTrimmed))
+      featureKeys.push(searchInputTrimmed);
```

Comment:
Are we sure about this ? Every time we type a letter, it will update the `featureKeys`, which will always contain at least the value of `searchInputTrimmed`. Then, inside `FeatureFlagDetails`, we call `getFeature` for this `searchInputTrimmed`, which will returns an error if the searched value does not exist (so for each letter we type, until we reach an existing feature flag, we will send requests that will fail). 

I don't think we have a choice here, because we need to search for "hidden" feature flags (otherwise, we could filter first on the `defaultFeatures` and then send the requests).

But maybe we should add a comment here:
```The search input is added to the `featureKeys` in order to check for "hidden" feature flags```